python/pyarrow/tests/test_compute.py

```diff
@@ -1316,6 +1318,67 @@ def test_arithmetic_multiply():
     assert result.equals(expected)
 
 
+@pytest.mark.parametrize("ty", ["round", "mround"])
+def test_round_to_integer(ty):
+    if ty == "round":
+        round = pc.round
+        RoundOptions = partial(pc.RoundOptions, ndigits=0)
+    elif ty == "mround":
+        round = pc.mround
+        RoundOptions = partial(pc.MRoundOptions, multiple=1)
+
+    values = [3.2, 3.5, 3.7, 4.5, -3.2, -3.5, -3.7, None]
+    rmode_and_expected_map = {
+        "down": [3, 3, 3, 4, -4, -4, -4, None],
+        "up": [4, 4, 4, 5, -3, -3, -3, None],
+        "towards_zero": [3, 3, 3, 4, -3, -3, -3, None],
+        "towards_infinity": [4, 4, 4, 5, -4, -4, -4, None],
+        "half_down": [3, 3, 4, 4, -3, -4, -4, None],
+        "half_up": [3, 4, 4, 5, -3, -3, -4, None],
+        "half_towards_zero": [3, 3, 4, 4, -3, -3, -4, None],
+        "half_towards_infinity": [3, 4, 4, 5, -3, -4, -4, None],
+        "half_to_even": [3, 4, 4, 4, -3, -4, -4, None],
+        "half_to_odd": [3, 3, 4, 5, -3, -3, -4, None],
+    }
+    for round_mode, expected in rmode_and_expected_map.items():
+        options = RoundOptions(round_mode=round_mode)
+        result = round(values, options=options)
+        assert np.all(np.isclose(result, pa.array(expected), equal_nan=True))
```

Comment:
Given the results are all integers, we should check for exact equality here, not approximate.

Reply:
Also, the Numpy [testing assertions](https://numpy.org/doc/stable/reference/routines.testing.html) should give a more informative output on failure ;-)